### PR TITLE
Getting mapped bam and bai published in the same folder

### PR DESCRIPTION
**Pull request by asp8200**
Issue #1540 

QUESTION : Should I add a pytest covering this bug? (It should be super easy to add such a test using the below-mentioned ntest-cmd.)

------------
The following test works on this PR:
```
nextflow run main.nf -profile test,alignment_to_fastq,docker,debug --outdir results --save_mapped --save_output_as_bam
````
and gives
```
├── mapped
│   └── test
│       ├── test.sorted.bam
│       └── test.sorted.bam.bai
```

I also tried fixing the problem by setting

https://github.com/nf-core/sarek/blob/b5b766d3b4ac89864f2fa07441cdc8844e70a79e/conf/modules/aligner.config#L52C21-L52C52

to `) { "mapped/${meta.sample}/${it}" }`, but then I got
```
results/preprocessing/mapped/<meta.sample>/<meta.id>.sorted.bam
results/preprocessing/mapped/<meta.sample>/<meta.id>.sorted.bam.bai
```
that is, the above-mentioned test gave:
```
├── mapped
│   └── test
│       ├── test-1.sorted.bam
│       └── test-1.sorted.bam.bai
```

I suggest that we add the above-mentioned test as a pytest.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test tests/ --verbose --profile +docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Changelog?

**Comment by asp8200** [comment]
> Changelog?

Sure, but what about adding a pytest covering this bug, so that we don't reintroduce the bug later?

**Comment by asp8200** [comment]
> > Changelog?
> 
> @maxulysse : Sure, but what about adding a pytest covering this bug, so that we don't reintroduce the bug later? How do we test that this is also working when `meta.size * meta.num_lanes != 1` ?

@maxulysse : I ran
```
nextflow run main.nf -profile test,docker,debug --outdir results --save_mapped --save_output_as_bam
```
where we - as expected - get `meta.num_lanes = 2` and also in that case we get published
```
├── mapped
│   └── test
│       ├── test.sorted.bam
│       └── test.sorted.bam.bai
```

**Review by maxulysse — DISMISSED** [review]

File: workflows/sarek/main.nf [hunk 1]
```diff
@@ -253,6 +253,12 @@ workflow SAREK {
             }
             .set { reads_grouping_key }
 
+        reads_for_alignment = reads_for_alignment.map{ meta, reads ->
+            // Update meta.id to meta.sample no multiple lanes or splitted fastqs
+            if (meta.size * meta.num_lanes == 1) [ meta + [ id:meta.sample ], reads ]
+            else [ meta, reads ]
+        }
+
```
Comment (FriederikeHanssen) [inline workflows/sarek/main.nf:261]: can this not be moved up to the previous mapping statement?

Comment (FriederikeHanssen) [reply workflows/sarek/main.nf:261]: or is it important that the mapping for the id is not stored in reads_grouping_key?

Comment (asp8200) [reply workflows/sarek/main.nf:261]: Not sure what you mean, but you're welcome to move stuff around.

Comment (FriederikeHanssen) [reply workflows/sarek/main.nf:261]: Never mind, the other is Adam's cool grouping logic. One more concern from my side, below there is the joining between the `reads_grouping_key` and `reads_alignment` meta's. Is this working for both splitting and unsplitting and across several lanes?

Comment (asp8200) [reply workflows/sarek/main.nf:261]: No idea 😬

You mean these combine-statements:

https://github.com/asp8200/sarek/blob/8d1592c242fbb2bd2cc7da66377032d55afc7816/workflows/sarek/main.nf#L270

https://github.com/asp8200/sarek/blob/8d1592c242fbb2bd2cc7da66377032d55afc7816/workflows/sarek/main.nf#L288

 I just hope we have CI-tests covering all this intricate stuff 🤞

Comment (maxulysse) [reply workflows/sarek/main.nf:261]: How about we merge as is for now, and we'll take care of beautifying it when we refactor?

**Review by maxulysse — APPROVED** [review]

**Pull request merged**